ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -187,18 +189,29 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
     public incrementStep(scroll: boolean = true): void {
         const nextIndex = this.nextAvailableSectionIndex();
         if (nextIndex !== -1) {
-            if (scroll) {
-                this.scrollToTop();
-            }
             // enable any validation errors to be visible
             this.validationRequested = true;
             this.sendSectionAnalytics();
             this.currentSection.validate();
+
+            if (scroll) {
+                this.scrollToTop();
+            }
+
             if (this.currentSection.valid) {
                 this.resetValidationState();
-                this.currentSectionIndex = nextIndex;
-                this.visitedSectionIndexes[nextIndex] = true;
-                this.saveLastVisitedSectionIndex(nextIndex);
+
+                this.isPageBusy.pipe(
+                    tap(_ => this.isLoaded$.next(false)),
+                    debounceTime(250),
+                    filter(busy => !busy),
```

Comment:
`isPageBusy` status changes several times in Mailing Address component (e.g. when saving address, removing temp address). 
Try to "calm down the waves" and get final value of `isPageBusy`  (as to get completed validation of  required mailing address ) before moving Next (to next section).